crates/credentials/src/presentation_definition_v2.rs

```diff
@@ -0,0 +1,230 @@
+#[serde(rename_all = "snake_case")]
+pub enum ConformantConsumerDisclosure {
+    Required,
+    Preferred,
+}
+
+/// Represents the format of a presentation definition
+///
+/// See `format` as defined in
+/// [Input Descriptor](https://identity.foundation/presentation-exchange/#input-descriptor-object)
+/// and [Registry](https://identity.foundation/claim-format-registry/#registry)
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Format {
+    pub jwt: Option<JwtObject>,
+    pub jwt_vc: Option<JwtObject>,
+    pub jwt_vp: Option<JwtObject>,
+}
+
+/// Represents a JWT object.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct JwtObject {
+    pub alg: Vec<String>,
+}
+
+/// Represents submission requirements for a presentation definition.
+#[skip_serializing_none]
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct SubmissionRequirement {
+    pub name: Option<String>,
+    pub purpose: Option<String>,
+    pub rule: Rule,
+    pub count: Option<u32>,
+    pub min: Option<u32>,
+    pub max: Option<u32>,
+    pub from: Option<String>,
+    pub from_nested: Option<Vec<SubmissionRequirement>>,
+}
+
+/// Enumeration representing presentation rule options.
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Rule {
+    #[default]
+    All,
+    Pick,
+}
+
+/// Enumeration representing optionality.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Optionality {
+    Required,
+    Preferred,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+    use std::fs;
+    use std::path::Path;
+
+    #[test]
+    fn can_serialize() {
+        let pd = PresentationDefinitionV2 {
+            id: "tests-pd-id".to_string(),
+            name: "simple PD".to_string().into(),
+            purpose: "pd for testing".to_string().into(),
+            input_descriptors: vec![InputDescriptorV2 {
+                id: "whatever".to_string(),
+                purpose: "purpose".to_string().into(),
+                constraints: ConstraintsV2 {
+                    fields: vec![FieldV2 {
+                        id: "field-id".to_string().into(),
+                        path: vec!["$.issuer".to_string()],
+                        purpose: "purpose".to_string().into(),
+                        filter: json!({"type": "string", "const": "123"}).into(),
+                        ..Default::default()
+                    }]
+                    .into(),
+                    limit_disclosure: Some(ConformantConsumerDisclosure::Required),
+                },
+                ..Default::default()
+            }],
+            ..Default::default()
+        };
+
+        let serialized_pd = serde_json::to_string(&pd).unwrap();
+
+        assert!(serialized_pd.contains("input_descriptors"));
+        assert!(serialized_pd.contains("123"));
```

Comment:
I'm not really convinced that this test fully tests serialization, but it's currently what we're doing in the [web5-kt](https://github.com/TBD54566975/web5-kt/blob/main/credentials/src/test/kotlin/web5/sdk/credentials/PresentationDefinitionTest.kt#L51-L52) test suite.

We could really use some test vectors here to have a more robust test of this functionality.

Reply:
I agree. I think most is already tested in the idempotency test.